themes/default/content/registry/packages/aws-native/_index.md

```diff
@@ -250,3 +250,144 @@ resources:
 {{% /choosable %}}
 
 {{% /chooser %}}
+
+## Third Party Resources
+
+The SDK for the AWS Native Cloud Control provider only includes resources for which Amazon have published the corresponding CloudFormation specifications. AWS also supports [CloudFormation extensions via the CloudFormation registry](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry.html) which can also be accessed via Cloud Control.
+
+If you want to manage resources using Pulumi's AWS Native Cloud Control Provider which aren't in the SDK, you can use the [ExtensionResource](https://www.pulumi.com/registry/packages/aws-native/api-docs/extensionresource/) resource within root of the SDK. The input properties and output properties are untyped allowing passing in any arbitrary values which will be passed to the Cloud Control API and tracked with Pulumi's managed state model.
+
+Here's a very simple demonstration of using the ExtensionResource to create an S3 bucket:
+
+{{< chooser language "typescript,python,go,csharp,java,yaml" / >}}
+
+{{% choosable language "javascript,typescript" %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws_native from "@pulumi/aws-native";
+
+const myBucket = new aws_native.ExtensionResource("myBucket", {
+    type: "AWS::S3::Bucket",
+    properties: {
+        BucketName: "my-bucket",
```

Comment:
Can we randomize the name? If I try this it says bucket already exists, which is likely.

Reply:
After updating the resource

```
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: f5ee580c-289f-47ec-ad3b-57ee50131f75, api error ValidationException: 2 validation errors detected: Value '' at 'typeName' failed to satisfy constraint: Member must have length greater than or equal to 10; Value '' at 'typeName' failed to satisfy constraint: Member must satisfy regular expression pattern: [A-Za-z0-9]{2,64}::[A-Za-z0-9]{2,64}::[A-Za-z0-9]{2,64}

```

Reply:
Randomising the name is the exact purpose of this PR which has been waiting on an approval for 3 weeks:
- https://github.com/pulumi/pulumi-aws-native/pull/1453

I'm not too sure where those validation errors are coming from ... this passed for me when testing.

Reply:
Ah I see, well the bucket naming issue is probably fine the users will figure it out, but should this like be typeName instead type? Why do I hit the error? I can try again.

Reply:
```python
import pulumi
import pulumi_aws_native as aws_native

my_bucket = aws_native.ExtensionResource("myBucket",
    type="AWS::S3::Bucket",
    properties={
        "BucketName": "my-bucket-t0yv0-2024apr18",
    })

```

This worked on `pulumi_aws_native  0.102.0`, perhaps I was fat-fingering something yesterday.